### DISCUSSION TITLE
Welcome to Flight-Booking-System Discussions!

### DISCUSSION THREAD

**Discussion by Nishant-Pandey-2004** [Nishant-Pandey-2004/Flight-Booking-System#2] [Q&A] [answered]
Since mongodb is used in this project, what are the limitations to it ?

**Comment by karan-panda** [comment, marked as answer]
### Limitations to consider when using MongoDB in your diabetes detection project:

1. Data Size and Scaling: MongoDB is great for most projects, but managing very large datasets can be tricky. You'll need to plan for scaling as your data grows.
2. Data Integrity: While MongoDB offers strong consistency within a single document, it may not provide full data integrity across multiple documents. Be cautious when handling critical health data.
3. Complex Queries: MongoDB is excellent for simple queries but might not be ideal for complex analytics. If your project involves complex data analysis, you might need additional tools.
4. Indexing: Make sure to set up proper indexes for efficient query performance. It's essential to optimize your database for your specific needs.
5. Schema Flexibility: MongoDB's flexible schema can be an advantage, but it can also lead to data inconsistencies if not managed carefully. Plan your data structure thoughtfully.
